causallib/time_varying/GFormula.py

```diff
@@ -0,0 +1,199 @@
+            ):
+        """
+            Fits parametric models and calculates prediction curves.
+
+            Args:
+                X (pd.DataFrame): Baseline covariate matrix of size (num_subjects, num_features).
+                a (pd.Series): Treatment assignment of size (num_subjects,).
+                t (pd.Series): Followup duration, size (num_subjects,).
+                y (pd.Series): Observed outcome (1) or right censoring event (0), size (num_subjects,).
+                refit_models (bool): if True, re-fit a the treatment model and covariate models.
+                kwargs (dict): Optional kwargs for fit call of survival model
+
+            Returns:
+                self
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def estimate_individual_outcome(self,
+                                    X: pd.DataFrame,
+                                    a: pd.Series,
+                                    t: pd.Series,
+                                    y: Optional[Any] = None,
+                                    timeline_start: Optional[int] = None,
+                                    timeline_end: Optional[int] = None
+                                    ) -> pd.DataFrame:
+        """
+              Returns individual estimated curves for each subject row in X/a/t
+
+              Args:
+                  X (pd.DataFrame): Baseline covariate matrix of size (num_subjects, num_features).
+                  a (pd.Series): Treatment assignment of size (num_subjects,).
+                  t (pd.Series): Followup durations, size (num_subjects,).
+                  y: NOT USED (for API compatibility only).
+                  timeline_start (int): Common start time-step. If provided, will generate survival curves starting
+                                        from 'timeline_start' for all patients.
+                                        If None, will predict from first observed event (t.min()).
+                  timeline_end (int): Common end time-step. If provided, will generate survival curves up to 'timeline_end'
+                                      for all patients.
+                                      If None, will predict up to last observed event (t.max()).
+
+              Returns:
+                  pd.DataFrame: with time-step index, subject IDs (X.index) as columns and ??
+              """
+        pass
+
+    @abc.abstractmethod
+    def estimate_population_outcome(self,
+                                    X: pd.DataFrame,
+                                    a: pd.Series,
+                                    t: pd.Series,
+                                    y: Optional[Any] = None,
+                                    timeline_start: Optional[int] = None,
+                                    timeline_end: Optional[int] = None
+                                    ) -> pd.DataFrame:
+        """
+              Returns individual estimated curves for each subject row in X/a/t
```

Comment:
@JulinaM  seems like the description for this and individual outcome is the same - please check